src/test/java/bio/terra/common/ValidationUtilsTest.java

```diff
@@ -76,4 +88,64 @@ public void testValidationOfValidInputString() {
     // no exception is thrown
     ValidationUtils.requireNotBlank("abc", "error msg");
   }
+
+  @Test
+  public void testRelationshipValidationDifferentDataType() {
+    RelationshipTermModel fromTerm = new RelationshipTermModel().column("hasCat").table("person");
+    RelationshipTermModel toTerm = new RelationshipTermModel().column("ownerId").table("car");
+    defineSampleTables();
+
+    LinkedHashMap errors =
+        ValidationUtils.validateMatchingColumnDataTypes(fromTerm, toTerm, tables);
+    assertThat("There should be one error.", errors.size(), equalTo(1));
```

Comment:
This could be a good place to use the `hasSize` matcher:
```
assertThat("There should be one error.", errors, hasSize(1));
```

It would be nice if there was an easy way to assert that the error itself is what we'd expect.  The way I think of it, if this test failed we might have a hard time debugging why (though that will be easier if using `hasSize` matcher, since we'll see `errors` emitted in output in the event the assertion fails).

Sometimes I'll put the error message in a `@VisibleForTesting` function and assert that the error matches.

Reply:
Same case as below, I don't think "hasSize" works on Maps without converting to a list. 

I've added another check that checks for the exact expected error message.